middleware/kubernetes/setup.go

```diff
@@ -185,17 +185,15 @@ func kubernetesParse(c *caddy.Controller) (*Kubernetes, error) {
 							zone: args[1],
 						})
 						continue
-					} else {
-						return nil, fmt.Errorf("Incorrect number of arguments for federation. Got %v, expect 2.", len(args))
 					}
-					return nil, c.ArgErr()
+					return nil, fmt.Errorf("incorrect number of arguments for federation, got %v, expected 2", len(args))
```

Comment:
This will break a unit test ... Error message strings are checked by setup_test.go .